runtime-sdk/src/modules/consensus/mod.rs

```diff
@@ -36,21 +36,38 @@ mod test;
 const MODULE_NAME: &str = "consensus";
 
 /// Parameters for the consensus module.
-#[derive(Clone, Debug, cbor::Encode, cbor::Decode)]
+#[derive(Clone, Debug, PartialEq, Eq, cbor::Encode, cbor::Decode)]
 pub struct Parameters {
     pub consensus_denomination: token::Denomination,
+    pub consensus_scaling_factor: u64,
 }
 
 impl Default for Parameters {
     fn default() -> Self {
         Self {
             consensus_denomination: token::Denomination::from_str("TEST").unwrap(),
+            consensus_scaling_factor: 1,
         }
     }
 }
 
+/// Errors emitted during rewards parameter validation.
+#[derive(Error, Debug)]
+pub enum ParameterValidationError {
+    #[error("consensus scaling factor set to zero")]
+    ZeroScalingFactor,
+}
+
 impl module::Parameters for Parameters {
-    type Error = ();
+    type Error = ParameterValidationError;
+
+    fn validate_basic(&self) -> Result<(), Self::Error> {
+        if self.consensus_scaling_factor == 0 {
```

Comment:
Should there also be a check for ensuring that the scaling factor is a power of 10? I guess there's technically nothing wrong with a non power of 10 scaling factor, but not sure if there's ever a reason for the weirdness.

Could also consider negative scaling factors, but this could also be added later if there's ever a need for it.